### DISCUSSION TITLE
[1.4.2] readValidArray - cannot read property

### DISCUSSION THREAD

**Discussion by querta** [Chimildic/goofy#38] [Помощь с ошибкой] [answered]
Не выполняются скрипты из-за ошибки:
`TypeError: Cannot read property 'hasOwnProperty' of undefined
    at readValidArray(library:666:22)
    at updatePlatformRecentTracks(library:598:25)
    at update(library:586:26)
    at updateRecentTracks_(library:17:18)`

Строки 665, 666:
`let items = Cache.read(filename);
if (items[0].hasOwnProperty('track')) {`

Строки 597, 598:
` function updatePlatformRecentTracks(recentTracks, filename, findNewPlayedMethod) {
  let fileItems = readValidArray(filename);`

**Comment by Chimildic** [comment, marked as answer]
Попробуйте [обновить до 1.4.3](https://script.google.com/d/1DnC4H7yjqPV2unMZ_nmB-1bDSJT9wQUJ7Wq-ijF4Nc7Fl3qnbT0FkPSr/edit?usp=sharing)
